src/state/wallets/actions.ts

```diff
@@ -62,6 +62,9 @@ export const loadWallets = (walletIndex?: number) => async (
       allWallets.length > 1
         ? [{ label: 'All wallets', balance: allWalletsBalance, preferredBalanceUnit: 'BTCV' }, ...allWallets]
         : allWallets;
+    // hack, we do this to clear the transactions for deleted wallets, needs proper refactor
+    dispatch(clearTransactions());
+
```

Comment:
can not we delete transactions for certain wallets on DeleteWalletScreen?

Reply:
we can and we will, but it needs more effort. I wanted to the quick fix here and real refactor on 3 keys version as I think there is no point for investing too much time on the 1 key version